src/components/lec6/lec6task4/index.js

```diff
@@ -0,0 +1,38 @@
+//lecture 6, task 4
+import React, { Component } from "react";
+
+export default class Generator extends React.Component{
+    constructor(props){
+      super(props);
+      this.state={
+            min : props.min,
+            max : props.max,
+            randomNumber : 0,
+            generated : false,
+      }
+    }
+
+    generate = () => {
+        this.setState({randomNumber: Math.floor(Math.random()*(1+this.state.max-this.state.min)+this.state.min)});
+        this.setState({generated: true});
+    }
+
+    shouldComponentUpdate(nextProps, nextState){
+      if (nextState.randomNumber % 2 !== 0)
+        return false;
+      else
+        return true;
+    }
```

Comment:
I remember in my example I used this life cycle method to avoid updating state but in this case it is unnecessary. Just do the check in your `generate` function. and if `nextState.randomNumber % 2 !== 0` then don't call `this.setState()`